truffle/contracts/VaultNotary/VaultNotary.sol

```diff
@@ -0,0 +1,233 @@
+            "whitelistedOnlyForHash, only allow vault owner or the whitelisted users to access");
+        _;
+    }
+
+    /** @dev Modifier for limiting the access to grant and revoke permissions
+      * Will check the message sender, only the owner of a vault can do the operation decorated
+      * @param vaultId bytes16 ID of the vault to check
+      */
+    modifier vaultOwnerOnly(bytes16 vaultId) {
+        require(msg.sender == notaryMapping[vaultId].vaultOwner, "can only access by vault owner");
+        _;
+    }
+
+    /** @notice If we upgrade the version of the contract, the old contract
+      *  will not be allowed to register new Vault anymore. However, the existing
+      *  Vaults registered in that contract should still be able to be updated
+      */
+    modifier notDeprecated() {
+        require(!isDeprecated, "This version of the VaultNotary contract has been deprecated");
+        _;
+    }
+
+    /** @notice This is function to register a vault, will only do the registration if a vaultId is not
+                registered before
+      * @dev It sets the msg.sender to the vault owner and set the update permission of the owner to true
+      * It calls the setVaultUir and setVaultHash to initialize those two records
+      * @param vaultId bytes16 VaultID to create, is the same as shipment ID in our system
+      * @param vaultUri string Vault URI to set
+      * @param vaultHash string  Vault hash to set
+      */
+    function registerVault(bytes16 vaultId, string calldata vaultUri, string calldata vaultHash)
+        external
+        notDeprecated
+    {
+        require(isNotRegistered(vaultId), "vault should not be registered, in registerVault");
+        notaryMapping[vaultId].vaultOwner = msg.sender;
+
+        //work around for if (vaultUri != "")
+        bytes memory tempStringBytes = bytes(vaultUri);
+        if (tempStringBytes.length != 0)
+            setVaultUri(vaultId, vaultUri);
+
+        tempStringBytes = bytes(vaultHash);
+        if (tempStringBytes.length != 0)
+            setVaultHash(vaultId, vaultHash);
+
+        emit VaultRegistered(msg.sender, vaultId);
+    }
+
+    /** @notice Sets the contract isDeprecated flag. Vault registration will be disabled if isDeprecated == True
+      * Only contract owner can set this
+      * @param _isDeprecated bool Boolean control variable
+      */
+    function setDeprecated(bool _isDeprecated)
+        external
+        onlyOwner
+    {
+        isDeprecated = _isDeprecated;
+        emit ContractDeprecatedSet(msg.sender, isDeprecated);
+    }
+
+    /** @notice Function to grant update permission to both Hash field in one vault
+      * @param vaultId bytes16 The ID of Vault to grant permission
+      * @param anotherAddress address The address to grant permission
+      */
+    function grantUpdateHashPermission(bytes16 vaultId, address anotherAddress)
```

Comment:
```suggestion
    function grantUpdateHashPermission(bytes16 vaultId, address addressToGrant)
```
something like this? @mlclay

Reply:
Yeah I think that's a little more clear as to the intent of that parameter.